lesson_2/mortgage_calc.py

```diff
@@ -0,0 +1,158 @@
+
+def error_output(message):
+    print(f"\n    !!!! {message} !!!!\n\n    -------TRY AGAIN!-------")
+
+def get_loan_amount():
+    prompt(MESSAGES["loan_amount"])
+    loan_amount_input = input()
+    return loan_amount_input
+
+def invalid_loan_amount(input_value):
+    try:
+        float(input_value)
+        if float(input_value) <= 0:
+            return True
+    except ValueError:
+        return True
+    return False
+
+def get_apr():
+    prompt(MESSAGES["APR"])
+    annual_percentage_rate_input = input()
+    return annual_percentage_rate_input
+
+def invalid_apr(input_apr):
+    try:
+        float(input_apr)
+        if float(input_apr) < 0:
+            return True
+    except ValueError:
+        return True
+    return False
+
+def get_loan_year_duration():
+    prompt(MESSAGES["years"])
+    year_duration = input()
+    while invalid_loan_duration(year_duration):
+        error_output(MESSAGES["invalid_year"])
+        prompt(MESSAGES["years"])
+        year_duration = input()
+    year_month_conversion = int(year_duration) * 12
+    return year_month_conversion
+
+def get_loan_month_duration():
+    prompt(MESSAGES["months"])
+    month_duration = input()
+    while invalid_loan_duration(month_duration):
+        error_output(MESSAGES["invalid_months"])
+        prompt(MESSAGES["months"])
+        month_duration = input()
+    month_duration = int(month_duration)
+    return month_duration
+
+def invalid_loan_duration(loan_duration):
+    try:
+        int(loan_duration)
+        if int(loan_duration) < 0:
+            return True
+    except ValueError:
+        return True
+    return False
+
+def zero_or_less_duration(year_month_conversion, month_duration):
+    duration = year_month_conversion + month_duration
+    if duration <= 0:
+        return True
+    return False
+
+def calc_again():
+    prompt(MESSAGES["another_calculation"])
+    another_calc = input()
+    if another_calc.lower() == 'y':
+        print('\n-------NEW CALCULATION-------')
+        return loan_calculator()
+
+    return intro_outro('SEE YOU LATER!')
+
+# LOAN CALC PROGRAM - uses functions above
+def loan_calculator():
+    # LOAN AMOUNT INPUT:
+    loan_amount_input = get_loan_amount()
+    while invalid_loan_amount(loan_amount_input):
+        error_output(MESSAGES["invalid_loan_amount"])
+        loan_amount_input = get_loan_amount()
+    loan_amount = float(loan_amount_input)
+
+    # MONTHLY APR:
+    annual_percentage_rate_input = get_apr()
+    while invalid_apr(annual_percentage_rate_input):
+        error_output(MESSAGES["invalid_apr"])
+        annual_percentage_rate_input = get_apr()
+    annual_percentage_rate = float(annual_percentage_rate_input)
+
+    if annual_percentage_rate != 0:
+        # CONVERT APR TO DECIMAL FORMAT
+        apr_decimal = float(annual_percentage_rate_input) * .01
+        # APR -> MONTHLY INTEREST RATE
+        monthly_interest_rate = apr_decimal / 12
+    else:
+        annual_percentage_rate = 'No APR'
```

Comment:
nit: I would use the `None` keyword here, it's built into python and would make your code more declarative.

example:

```py
annual_percentage_rate = None
```

Then you would do:
```py
if annual_percentage_rate is None:
  #...
```

or:
```py
if annual_percentage_rate is not None:
  #...
```